src/EventStore.SourceGenerators/Messaging/Attributes.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace EventStore.SourceGenerators.Messaging {
+
+	public class RootMessageAttribute : Attribute {
```

Comment:
From a consumption perspective it's better to have the attributes defined in the "main"-code, not generated.

E.g. when you start a new project, then there's no attribute before the generator creates it. That can cause troubles with VS (even today).
Thus the common pattern (at least in dotnet-org) seems to be to have the needed attributes be parts of the shipped binaries, so the generator just "consumes" them.

Reply:
thanks! ive adjusted this